src/metabase/embed/settings.clj

```diff
@@ -31,39 +38,67 @@
   :visibility :authenticated
   :export?    true
   :audit      :getter
-  :setter     (fn [new-value]
-                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding))
-                  (setting/set-value-of-type! :boolean :enable-embedding new-value)
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32)))
-                  (snowplow/track-event! ::snowplow/embed_share
-                                         {:event                      (if new-value
-                                                                        :embedding-enabled
-                                                                        :embedding-disabled)
-                                          :embedding-app-origin-set   (boolean (embedding-app-origin))
-                                          :number-embedded-questions  (t2/count :model/Card :enable_embedding true)
-                                          :number-embedded-dashboards (t2/count :model/Dashboard :enable_embedding true)}))))
+  :deprecated "0.51.0"
+  :setter     (make-embedding-toggle-setter :enable-embedding "embedding"))
 
-(defsetting enable-embedding-static
-  (deferred-tru "Allow admins to embed Metabase via static embedding?")
+(defsetting embedding-app-origin
+  (deferred-tru "Allow this origin to embed the full Metabase application.")
+  ;; This value is usually gated by [[enable-embedding]]
+  :feature    :embedding
+  :type       :string
+  :export?    false
+  :visibility :public
+  :audit      :getter)
+
+(defsetting enable-embedding-sdk
+  (deferred-tru "Allow admins to embed Metabase via the SDK?")
   :type       :boolean
   :default    false
   :visibility :authenticated
   :export?    false
   :audit      :getter
-  :setter     (fn [new-value]
-                (when (not= new-value (setting/get-value-of-type :boolean :enable-embedding-static))
-                  (setting/set-value-of-type! :boolean :enable-embedding-static new-value)
-                  (when (and new-value (str/blank? (embed/embedding-secret-key)))
-                    (embed/embedding-secret-key! (crypto-random/hex 32))))))
+  :setter     (make-embedding-toggle-setter :enable-embedding-sdk "sdk-embedding"))
 
-(defsetting enable-embedding-sdk
-  (deferred-tru "Allow admins to embed Metabase via the SDK?")
+(mu/defn- ignore-localhost :- :string
+  "Remove localhost:* or localhost:<port> from the list of origins."
```

Comment:
Is it an issue that we're not also removing `127.0.0.1` or other potential representations of localhost? Or would that not matter?

Reply:
I also wonder, do we actually need this `ignore-localhost` logic? Isn't the important thing just that it's _added_ when fetching the setting value? (I might be totally misunderstanding the requirements here though)

Reply:
It's kind of subtle, but I think we want to ignore it because if you keep adding localhost:x, then we don't want the actual value to keep growing into `"localhost:* localhost:1000 localhost:3000 ..."`.

Reply:
Especially since the getter always prefixes the value with "localhost:*", meaning if you set the setting N times, e.g. from the UI, you'd get N "localhost:* " prefixes.